clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	batch "k8s.io/api/batch/v1"
+	"k8s.io/api/core/v1"
+	extensions "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	defaultWaitForRCPodsTimeout      = 60 * time.Second
+	defaultWaitForRCPodsInterval     = 5 * time.Second
+	waitForControlledPodsRunningName = "WaitForControlledPodsRunning"
+)
+
+func init() {
+	measurement.Register(waitForControlledPodsRunningName, createWaitForControlledPodsRunningMeasurement)
+}
+
+func createWaitForControlledPodsRunningMeasurement() measurement.Measurement {
+	return &waitForControlledPodsRunningMeasurement{}
+}
+
+type waitForControlledPodsRunningMeasurement struct{}
+
+// Execute waits until all specified RCs have all pods running or until timeout happens.
+// RCs can be specified by field and/or label selectors.
+// If namespace is not passed by parameter, all-namespace scope is assumed.
+func (*waitForControlledPodsRunningMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
```

Comment:
Can you refactor this function into at least 2 parts (instead of single long one) - one for getting list of RCs and one for waiting for them to run?

Reply:
Done.